### PR TITLE
Pars logs

### PR DESCRIPTION
**Pull request by kolya182**
Switching to parse transaction input data from logs.

### CODE REVIEW

File: src/common/TokenParser.ts [hunk 1]
```diff
@@ -1,43 +1,42 @@
...
-        // extract  valid contracts
-        let contractAddresses: any = [];
         transactions.map((transaction: any) => {
-            const decodedInput = this.abiDecoder.decodeMethod(transaction.input);
-            if (
-                (decodedInput && decodedInput.name === "transfer" && Array.isArray(decodedInput.params) && decodedInput.params.length === 2 && transaction.to !== null) ||
-                (decodedInput && decodedInput.name === "mint" && Array.isArray(decodedInput.params) && decodedInput.params.length === 2 && transaction.to !== null)
-            ) {
-                contractAddresses.push(transaction.to.toLowerCase());
+            const decodedLogs = this.abiDecoder.decodeLogs(transaction.receipt.logs);
```
Comment (michaelScoff) [inline src/common/TokenParser.ts]: can you call `flatMap` to avoid checking for `if(log)`?

File: src/common/TokenParser.ts [hunk 2]
```diff
@@ -1,43 +1,42 @@
...
+                        }
+                    }
+                });
             }
         });
-        // remove duplicates
-        contractAddresses = contractAddresses.filter((elem: any, pos: any, arr: any) => arr.indexOf(elem) == pos);
-        // process contracts
+
+        contractAddresses = contractAddresses.filter((elem: any, pos: any, arr: any) => arr.indexOf(elem) == pos);;
```
Comment (michaelScoff) [inline src/common/TokenParser.ts]: Can you use `Set` in here to keep unique contracts?

File: src/common/TransactionParser.ts [hunk 3]
```diff
@@ -51,112 +47,139 @@ export class TransactionParser {
...
-                if (decodedInput && decodedInput.name === "transfer" && Array.isArray(decodedInput.params) && decodedInput.params.length == 2) {
-                    const to = decodedInput.params[0].value.toLowerCase();
-                    const value = removeScientificNotationFromNumbString(decodedInput.params[1].value);
-                    const p = this.findOrCreateTransactionOperation(transaction._id, transaction.from, to, value, contract._id);
-                    operationPromises.push(p);
+            const decodedLogs = erc20ABIDecoder.decodeLogs(transaction.receipt.logs);
+                // Covers where contract https://etherscan.io/tx/0x83388873b5a6d2d462e1f61e09dfcc6df779e2da46c8a1727f76e0e68b4c5ecf
+                if (contract) {
+                    if (decodedLogs.length > 0) {
+                        console.log("decodedLogs", decodedLogs)
```
Comment (michaelScoff) [inline src/common/TransactionParser.ts]: Remove console.log and other comments here.

File: src/common/TransactionParser.ts [hunk 4]
```diff
@@ -51,112 +47,139 @@ export class TransactionParser {
...
-                return Promise.resolve(operation);
-            }).catch((err: Error) => {
-                winston.error(`Could not add operation to transaction with ID ${transactionId} with error: ${err}`);
-            });
-        }).catch((err: Error) => {
-            winston.error(`Could not save transaction operation with error: ${err}`);
-        });
+    }
+
+    private isLogValid(log: {name: string, events: any[]}) {
```
Comment (michaelScoff) [inline src/common/TransactionParser.ts]: unused `isLogValid `

File: src/common/operations/LocalizedOperationConverter.ts [hunk 5]
```diff
@@ -9,18 +9,22 @@ export class LocalizedOperationConverter {
...
-            "title": `Transfer ${contract.symbol}`,
-            "type": operation.type,
-            "from": operation.from,
-            "to": operation.to,
-            "contract": contract.address,
-            "value": String(value),
-            "new_value": operation.value,
-            "symbol": contract.symbol,
-            "decimals": contract.decimals
+        if (operation.contract) {
```
Comment (michaelScoff) [inline src/common/operations/LocalizedOperationConverter.ts]: Why do we check contract in here? if we always expect to be there.

File: src/common/TransactionParser.ts (hunk 4 above)
Comment (kolya182) [reply src/common/TransactionParser.ts]: Fixed

File: src/common/TransactionParser.ts (hunk 3 above)
Comment (kolya182) [reply src/common/TransactionParser.ts]: Fixed

File: src/common/TokenParser.ts (hunk 2 above)
Comment (kolya182) [reply src/common/TokenParser.ts]: Fixed

File: src/common/TokenParser.ts [hunk 6]
```diff
@@ -117,7 +113,7 @@ export class TokenParser {
 
     private convertSymbol(symbol: string): string {
         if (symbol.startsWith("0x")) {
-            return Config.web3.utils.hexToAscii(symbol);
+            return Config.web3.utils.hexToAscii(symbol).replace(/\u0000*$/, "");
```
Comment (michaelScoff) [inline src/common/TokenParser.ts]: what's `u0000*$`?

File: src/controllers/WalletsController.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import { Wallet } from "../models/WalletModel";
```
Comment (michaelScoff) [inline src/controllers/WalletsController.ts]: what's this doing here in PR?

File: src/common/TokenParser.ts (hunk 6 above)
Comment (kolya182) [reply src/common/TokenParser.ts]: Removes screenshot 2018-01-26 20 14 00 Example of trx https://api.trustwalletapp.com/transactions/0x34aaa2068d7a9609af5f8618aa83ab0129985f794ebf05d3d1be3c876e1c78d5

File: src/common/TokenParser.ts (hunk 1 above)
Comment (kolya182) [reply src/common/TokenParser.ts]: @michaelScoff I used `filter` just less code require, you right `if(log)` not necessary anymore. Fixed.

**Pull request merged**